lib/internal/test_runner/runner.js

```diff
@@ -382,7 +382,7 @@ function runTestFile(path, filesWatcher, opts) {
       filesWatcher.runningSubtests.delete(path);
       if (filesWatcher.runningSubtests.size === 0) {
         opts.root.reporter[kEmitMessage]('test:watch:drained');
-        queueMicrotask(() => opts.root.postRun());
+        SafePromisePrototypeFinally(subTestEnded, () => opts.root.postRun());
```

Comment:
I would not recommend using `SafePromisePrototypeFinally`, it's meant to be used when the resulting promise is exposed to the user – in this case, wrapping in a `SafePromise` is pure overhead.
```suggestion
        queueMicrotask(async () => {
          try {
            await subTestEnded;
          } catch {
            // Potential errors should already have been reported to the user.
          } finally {
            opts.root.postRun();
          });
```